### ISSUE TITLE
Preferences dialog is mega stretched and won't shrink

### ISSUE THREAD

**Issue by mgrandi** [sqlitebrowser/sqlitebrowser#1755]
Details for the issue
--------------------

#### What did you do?
go to "Db Browser for Sqlite -> preferences"

#### What did you expect to see?
a normal preferences window

#### What did you see instead?

a very _wide_ preferences window that makes it pretty much impossible to see the actual settings

![screenshot_1152](https://user-images.githubusercontent.com/802786/53131985-8bfd2600-352b-11e9-8349-08a51c02d438.png)

![screenshot_1153](https://user-images.githubusercontent.com/802786/53132016-9b7c6f00-352b-11e9-8a5e-5abae158c256.png)

Useful extra information
-------------------------

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

> DB4S v3.11.1 [built for x86_64-little_endian-lp64] on macOS 10.14 (darwin/18.2.0) [x86_64]
> using SQLCipher Version 4.0.1 community (based on SQLite 3.26.0)
> and Qt 5.11.3

(also tried: 

```
Version 3.11.99 (Feb 20 2019)

Built for x86_64-little_endian-lp64, running on x86_64

Qt Version 5.11.3

SQLCipher Version 4.0.1 community (based on SQLite 3.26.0)
```

#### Did you also

- [X] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [X] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by justinclift** [comment]
Wow, this is a strange bug.  Um... if you try this build (created a few minutes ago, using a different release of Qt), does it work better?

&nbsp; &nbsp; https://nightlies.sqlitebrowser.org/osx-onceoffs/DB%20Browser%20for%20SQLite-sqlcipher_issue1751v1.dmg

Next thought, are you using a HiDPI monitor?  Kind of wondering if maybe there's some wrong screen size/resolution thing happening, so it's doubling the needed size. :confused:

**Comment by mgrandi** [comment]
while it looks better (better dark mode support), the prefs pane is still stretched

i am using 2 monitors, one that is just 1080p and the other is the retina laptop screen, but when i tried dragging DB4S to the laptop screen and opening the preferences, its still really wide. 

The preferences pane is also way more than 2x wider than it should be, so i don't think its a case of the DPI scale being off by a factor of 2, if that helps at all

**Comment by justinclift** [comment]
Thanks @mgrandi.  No idea at all how to go about diagnosing what's wrong here.

@revolter any ideas? :smile:

**Comment by mgrandi** [comment]
i looked up how to figure out the resolution since mac hides it from you in the "Display" preference panel, but this is what it says in `system report.app`

```
Intel Iris:

  Chipset Model:	Intel Iris
  Type:	GPU
  Bus:	Built-In
  VRAM (Dynamic, Max):	1536 MB
  Vendor:	Intel
  Device ID:	0x0a2e
  Revision ID:	0x0009
  Metal:	Supported, feature set macOS GPUFamily1 v4
  Displays:
Color LCD:
  Display Type:	Built-In Retina LCD
  Resolution:	2560 x 1600 Retina
  Framebuffer Depth:	24-Bit Color (ARGB8888)
  Mirror:	Off
  Online:	Yes
  Rotation:	Supported
  Automatically Adjust Brightness:	No
VA2702w:
  Resolution:	1920 x 1080 (1080p FHD - Full High Definition)
  UI Looks like:	1920 x 1080 @ 60 Hz
  Framebuffer Depth:	24-Bit Color (ARGB8888)
  Display Serial Number:	SC7103402975
  Main Display:	Yes
  Mirror:	Off
  Online:	Yes
  Rotation:	Supported
  Automatically Adjust Brightness:	No

```

**Comment by justinclift** [comment]
Thanks @mgrandi, that might help.

As a workaround for now, the only thing that's springing to mind is maybe try the previous release?

&nbsp; &nbsp; https://github.com/sqlitebrowser/sqlitebrowser/releases/tag/v3.10.1


It seems ok with macOS for other issues - eg Dark Mode on Mojave isn't completely busted - so may work better with this too.

**Comment by mgrandi** [comment]
yeah thats fine, let me know if you want me to test any more builds

**Comment by mgrojo** [comment]
@mgrandi, first of all, could you please make a screenshot of every tab? Maybe a hint appears.

And some questions so we can start to investigate it. Have you modified the preferences? Would it be possible to restore defaults and see whether it affects the window? Can you manually reduce the window size using the border handle?

**Comment by mgrandi** [comment]
on resizing: I can make it taller and wider, but not smaller than the default (the default height/width that it appears as when i go to "DB Browser for SQLite -> preferences"

also, i guess the app i use to take screenshots can take a picture of the entire window even if part of it is off the screen, neat! so i guess the resolution of the preferences window is `637px tall by 11466px tall` on my external monitor

![screenshot_1161](https://user-images.githubusercontent.com/802786/53268388-27fd6d80-36a3-11e9-9801-3d29592bf280.png)
![screenshot_1160](https://user-images.githubusercontent.com/802786/53268389-27fd6d80-36a3-11e9-94a4-b5e69fae5f57.png)
![screenshot_1159](https://user-images.githubusercontent.com/802786/53268390-27fd6d80-36a3-11e9-8efd-0d0d0ee6ec6b.png)
![screenshot_1158](https://user-images.githubusercontent.com/802786/53268391-28960400-36a3-11e9-9cb8-45eeb74b042c.png)
![screenshot_1157](https://user-images.githubusercontent.com/802786/53268392-28960400-36a3-11e9-9337-7fb06cc8facf.png)
![screenshot_1156](https://user-images.githubusercontent.com/802786/53268393-28960400-36a3-11e9-893c-06714ec58d21.png)

**Comment by justinclift** [comment]
@jsejcksn Probably a stretch... but would you have any idea what could possibly be going wrong here?  I'm at a loss to even know where to begin with this one. :frowning:

**Comment by jsejcksn** [comment]
@justinclift "A stretch" 🙄

@mgrandi Try trashing the prefs and restarting. I've read about this happening in other apps due to unhandled exceptions caused by "invalid" character combinations in `plist` files.

``` shell
rm ~/Library/Preferences/net.sourceforge.sqlitebrowser.plist
```

That's all I can contribute, so 🤞

**Comment by justinclift** [comment]
Thanks heaps @jsejcksn, that's more help than I've been so far. :smile:

**Comment by mgrandi** [comment]
I tried deleting `~/Library/Preferences/net.sourceforge.sqlitebrowser.plist` and `~/Library/Saved Application State/net.sourceforge.sqlitebrowser.savedState` and restarted the app. I'll try restarting the entire computer later today to see if that fixes anything

**Comment by justinclift** [comment]
@mgrandi It turns out there's two preference files (a left over legacy thing), not just one:

```
~/Library/Preferences/com.sqlitebrowser.sqlitebrowser.plist
~/jc/Library/Preferences/net.sourceforge.sqlitebrowser.plist
```

Probably need to kill both of those to test this potential fix. :smile:

**Comment by mgrandi** [comment]
no good =( 

```
[2019-02-26 11:03:18] markgrandi@Gypaetus:~$ ls ~/Library/Preferences/com.sqlitebrowser.sqlitebrowser.plist
/Users/markgrandi/Library/Preferences/com.sqlitebrowser.sqlitebrowser.plist
[2019-02-26 11:03:32] markgrandi@Gypaetus:~$ ls ~/Library/Preferences/net.sourceforge.sqlitebrowser.plist
ls: /Users/markgrandi/Library/Preferences/net.sourceforge.sqlitebrowser.plist: No such file or directory
[2019-02-26 11:03:44] markgrandi@Gypaetus:~$ rm  ~/Library/Preferences/com.sqlitebrowser.sqlitebrowser.plist
```

restarted and still get the stretched window

**Comment by mgrojo** [comment]
Is this using the version having problems with the dark mode? I ask because there are text fields apparently empty, but might have white foreground over white background. Have you tested with the [build that works around the dark mode](https://download.sqlitebrowser.org/DB.Browser.for.SQLite-3.11.1v2.dmg)?

**Comment by mgrandi** [comment]
with that version that @mgrojo posted, i still get the mega stretched preferences window

**Comment by jsejcksn** [comment]
@mgrandi Try this:

  1. Disconnect any external displays and don't reconnect them
  2. Quit DB Browser for SQLite
  3. Run the following:

``` shell
rm ~/Library/Preferences/com.sqlitebrowser.sqlitebrowser.plist
rm ~/Library/Preferences/net.sourceforge.sqlitebrowser.plist
rm -r ~/Library/Saved\ Application\ State/net.sourceforge.sqlitebrowser.savedState
```

Then, if you launch the app again, do you still have the issue?

**Comment by mgrandi** [comment]
just tried it, restarted my laptop and didn't connect a monitor, and it still happens both with the latest stable and one of the nightly builds

I am noticing that it looks like it is starting out as the original size and then is resizing to be really big,maybe that is a clue?

I can also try compiling this myself and see if i can somehow put a breakpoint when the window is being created to see why it thinks it needs to resize to be super wide

**Comment by justinclift** [comment]
> I can also try compiling this myself and see if i can somehow put a breakpoint when the window is being created to see why it thinks it needs to resize to be super wide

It's worth a shot.  If you can figure out why it thinks it needs to resize, we're definitely up for assisting. :smile:

Also, when you get close to finding the right spot, screenshot the heck out of everything.  Not need to post them all, but the screenshots generally turn out to be useful for weird problems, as it gives something you can go back and refer to while investigating. :smile:

**Comment by revolter** [comment]
I have no idea :( It never happened for me.

**Comment by justinclift** [comment]
Yeah, it's definitely one of the more unusual things.  It'd be good to figure out what's causing it. :smile:

**Comment by jsejcksn** [comment]
Re: screenshots: I have also found them to be assets during analysis. If you are hesitant to store them locally (because they don't fit in your organizational structure, etc.), you can utilize many free image hosting services (or perhaps this repo might afford you a specific issue thread for them—ask one of the members). @mgrandi @justinclift

**Comment by justinclift** [comment]
When doing screenshots myself, generally I throw them in a local folder (lots of screenshots).  As I hit relevant points of investigation and need to share info, then I'll pick the right screenshots to illustrate a point/problem/etc and paste them directly into the issue.  Seems to work ok. :smile:

**Comment by mgrojo** [comment]
@mgrandi 
> I am noticing that it looks like it is starting out as the original size and then is resizing to be really big,maybe that is a clue?

Then a video capture might help. Pausing could help then to notice any important hint that was too quick to be noticed to the naked eye.
